### PR TITLE
37: User Edits with Prompts {{}}

### PR DESCRIPTION
**Pull request by rnukala1982**
To fix user edits with {{}}. Planning to remove the handleSelectPromoptWithVariables

### CODE REVIEW

**Comment by rnukala1982** [comment]
Sample Working Video after the fix for reference: 

https://github.com/writingmate/chatlabs/assets/64998012/5611b18b-e823-476b-843f-d465271207c9

**Comment by vood** [comment]
@rnukala1982 thanks for the code `handleSelectPromoptWithVariables` is actually important code that triggers a popup for prompts with templates on selection. The bug here is that here is that any click inside the `SheetContent` of `sidebar-update-item.tsx` propagates up to the parent `SidebarItem` and triggers `handleSelectPromoptWithVariables` which should not be happening. 

A proper fix would be to stop `Click` event propagation on `SheetContent` and keep `handleSelectPromoptWithVariables` unchanged.

**Comment by rnukala1982** [comment]
You meant something like this @vood : https://github.com/writingmate/chatlabs/pull/42/commits/edd5828fc588127a5ea8d2c981326d78ec277ca0#diff-7a41b2006da2378b5a195b59e880268aec9175f7dfad916e8c80eecf049b4050

**Comment by vood** [comment]
No, I mean you don't need to comment handleSelectPromptWithVariables or modify it at all. 

The reason why the bug is happening is because `handleClickAction` gets triggered when it shouldn't. To prevent this from happening you need to stop onClick propagation inside `SidebarUpdateItem` on `SheetContent` element.

**Comment by rnukala1982** [comment]
Ok

**Comment by rnukala1982** [comment]
Updated based on the feedback.

File: components/sidebar/items/all/sidebar-display-item.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ export const SidebarItem: FC<SidebarItemProps> = ({
     chats: async (item: any) => {},
     presets: async (item: any) => {},
     prompts: async (item: any) => {
-      handleSelectPromptWithVariables(item)
+      handleSelectPrompt(item)
```
Comment (vood) [inline components/sidebar/items/all/sidebar-display-item.tsx:102]: Let's keep `handleSelectPromptWithVariables`, no need to change it to `handleSelectPrompt`

Comment (rnukala1982) [reply components/sidebar/items/all/sidebar-display-item.tsx]: updated it.

**Pull request merged**